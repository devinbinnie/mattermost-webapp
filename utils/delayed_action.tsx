// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default class DelayedAction {
    private action: Function;
    private timer: number;

    constructor(action: Function) {
        this.action = action;

        this.timer = -1;

        // bind fire since it doesn't get passed the correct this value with setTimeout
        this.fire = this.fire.bind(this);
    }

    fire() {
        this.action();

        this.timer = -1;
    }

    fireAfter(timeout: number) {
        if (this.timer >= 0) {
            window.clearTimeout(this.timer);
        }

        this.timer = window.setTimeout(this.fire, timeout);
    }

    cancel() {
        window.clearTimeout(this.timer);
    }
}
