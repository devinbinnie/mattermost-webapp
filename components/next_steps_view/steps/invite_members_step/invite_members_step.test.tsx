// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {TestHelper} from 'utils/test_helper';

import InviteMembersStep from './invite_members_step';

describe('components/next_steps_view/steps/invite_members_step', () => {
    const baseProps = {
        id: 'invite_members_step',
        teamId: 'team_id',
        onSkip: jest.fn(),
        onFinish: jest.fn(),
        currentUser: TestHelper.getUserMock(),
        maxFileSize: 1000000000,
        actions: {
            sendEmailInvitesToTeamGracefully: jest.fn(),
        },
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <InviteMembersStep {...baseProps}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
