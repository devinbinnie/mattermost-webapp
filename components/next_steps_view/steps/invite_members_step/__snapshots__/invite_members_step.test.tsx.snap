// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/next_steps_view/steps/invite_members_step should match snapshot 1`] = `
<div
  className="NextStepsView__stepWrapper"
>
  <div
    className="InviteMembersStep"
  >
    <div
      className="InviteMembersStep__emailInvitations"
    >
      <h3>
        <FormattedMessage
          defaultMessage="Send invitations via email"
          id="next_steps_view.invite_members_step.sendInvitationsViaEmail"
          values={Object {}}
        />
      </h3>
      <FormattedMessage
        defaultMessage="You can invite up to 10 team members using a space or comma between addresses"
        id="next_steps_view.invite_members_step.youCanInviteUpTo"
        values={Object {}}
      />
      <MultiInput
        inputValue=""
        legend="Email addresses"
        onChange={[Function]}
        onInputChange={[Function]}
        placeholder="Enter email addresses"
        styles={
          Object {
            "control": [Function],
            "placeholder": [Function],
            "valueContainer": [Function],
          }
        }
        value={Array []}
      />
      <div
        className="InviteMembersStep__send"
      >
        <button
          className="InviteMembersStep__sendButton disabled"
          disabled={true}
          onClick={[Function]}
        >
          <i
            className="icon icon-send"
          />
          <FormattedMessage
            defaultMessage="Send"
            id="next_steps_view.invite_members_step.send"
            values={Object {}}
          />
        </button>
        <div
          className="InviteMembersStep__invitationResults"
        />
      </div>
    </div>
    <div
      className="InviteMembersStep__shareInviteLink"
    >
      <h3>
        <FormattedMessage
          defaultMessage="Or share this link to invite members"
          id="next_steps_view.invite_members_step.orShareThisLink"
          values={Object {}}
        />
      </h3>
      <div
        className="InviteMembersStep__shareLinkBlock"
      >
        <input
          aria-label={
            Object {
              "defaultMessage": "team invite link",
              "id": "next_steps_view.invite_members_step.shareLinkInput",
            }
          }
          className="InviteMembersStep__shareLinkInput"
          data-testid="shareLinkInput"
          readOnly={true}
          type="text"
          value="http://localhost:8065/signup_user/"
        />
        <button
          className="InviteMembersStep__shareLinkInputButton"
          data-testid="shareLinkInputButton"
          onClick={[Function]}
        >
          <i
            className="icon icon-link-variant"
          />
          <FormattedMessage
            defaultMessage="Copy Link"
            id="next_steps_view.invite_members_step.copy_button"
            values={Object {}}
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="NextStepsView__wizardButtons"
  >
    <button
      className="NextStepsView__button confirm"
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Finish"
        id="next_steps_view.invite_members_step.finish"
        values={Object {}}
      />
    </button>
  </div>
</div>
`;
